code/game/objects/items/weapons/clown_items.dm

```diff
@@ -134,4 +134,35 @@
 	desc = "Damn son, where'd you find this?"
 	icon_state = "air_horn"
 	honksound = 'sound/items/AirHorn2.ogg'
-	cooldowntime = 50
\ No newline at end of file
+	cooldowntime = 50
+
+/obj/item/weapon/bikehorn/golden
+	name = "golden bike horn"
+	desc = "Flippin' awesome!"
+	icon_state = "gold_horn"
+	item_state = "gold_horn"
+
+/obj/item/weapon/bikehorn/golden/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(!spam_flag)
+		for(M in ohearers(7, user))
+			if(istype(M, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = M
+				if((istype(H.ears, /obj/item/clothing/ears/earmuffs)) || H.ear_deaf)
+					continue
+			M.emote("flip")
+	return ..()
+
+/obj/item/weapon/bikehorn/golden/attack_self(mob/living/carbon/M, mob/user)
+	if(!spam_flag)
+		spam_flag = 1
+		playsound(src.loc, honksound, 50, 1)
+		src.add_fingerprint(user)
+		for(M in ohearers(7, user))
+			if(istype(M, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = M
+				if((istype(H.ears, /obj/item/clothing/ears/earmuffs)) || H.ear_deaf)
+					continue
+			M.emote("flip")

```

Comment:
Put all this shit into a new proc /obj/item/weapon/bikehorn/golden/proc/flip_mobs()

Then call flip_mobs() instead of copypasting this whole loop in both attack and attack_self please